tests/ztest/base/src/main.c

```diff
@@ -6,11 +6,13 @@
 
 #include <ztest.h>
 
-static void test_empty_test(void)
+ZTEST_SUITE(framework_tests, NULL, NULL, NULL, NULL, NULL);
```

Comment:
I don't like that much that you need to repeat suite name for each `ZTEST` so wondering if there could be an option to have default suite. The most user friendly option that comes to my mind is to use fixed define like `ZTEST_DEFAULT_SUITE` and flip parameters order in `ZTEST`. Then turn `ZTEST` to variable arguments macro. Then you can detect if only one argument is used and use `ZTEST_DEFAULT_SUITE` as suite name then.

Test would look like:
```
#define ZTEST_DEFAULT_SUITE foo_suite
ZTEST_SUITE(ZTEST_DEFAULT_SUITE, ...)

ZTEST(test_empty)
{

}

ZTEST(foo_test)
{

}

Reply:
This is pretty consistent with other testing frameworks (gtest for example) but I see your point, let me think about that some more and see if I can come up with a proposal. I would assume that doing a default suite would be similar to how drivers are done (which would require a new macro such as defining `DT_DRV_COMPAT` and using the `DT_INST_` macros.

Reply:
OK, after thinking about this for a day, what I've come up with is using something like:
```
COND_CODE_0(
  NUM_VA_ARGS_LESS_1(__VA_ARGS__),
  (/* only 1 argument, use default suite */),
  (/* 2 arguments, use GET_ARG_N(1, __VA_ARGS__) to get the suite name */)
)
```

Though I'm not entirely convinced this is much better, I think it makes the code a bit more error prone. I would much prefer to have to explicitly state that a unit test is a part of a suite.

Reply:
yes, that macro set can be used to get that. Is that error prone? Hard to say. Compilation will fail when `ZTEST_DEFAULT_SUITE` is not defined. You can also add protection against passing more than 2 arguments
`BUILD_ASSERT(NUM_VA_ARGS_LESS_1(__VA_ARGS__) < 2);`.

Reply:
Would you mind if this was done in a following PR? It seems that this would be a superset of the current feature set. So I would prefer to get this change in and then have a discussion about allowing a `DEFAULT_TEST_SUITE` and extending the macro to optionally include the test suite's name. Would that be OK?

Reply:
yes, sounds good.